### DISCUSSION TITLE
get_purchase_orders_status not working for some countries

### DISCUSSION THREAD

**Discussion by timmyscj** [saleweaver/python-amazon-sp-api#428] [Q&A] [answered]
get_purchase_orders_status is working fine for me for US and CA but it is not working for non North American countries.

Is this an issue? All other reports are working fine for all countries.

[{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]

**Comment by saleweaver** [comment, marked as answer]
For all Auth related questions, please open a case with seller support. It doesn't sound like this is something this client could influence.

**Reply by timmyscj** [reply]
The exact same get_purchase_orders_status code works correctly for countries in North America but does not work for other regions. My credentials are correct because I am able to use Postman to request get_purchase_orders_status data. 

I am trying to work with Amazon support but I am not getting anywhere. I would truly appreciate any advice or potential solutions. I am at a dead end.

**Reply by timmyscj** [reply]
More details...

The code below works for Catalog Items report but gives an 'Invalid Signature' error when requesting the purchaseOrdersStatus report (commented out below). My credentials are correct because it works correctly when requesting purchaseOrderStatus via Postman. 

Can anyone provide and information or solutions? I am at a dead end. 

`import boto3
from requests_auth_aws_sigv4 import AWSSigV4
import requests
import utils.report_credentials as rc
import yaml

acct_credentials = yaml.safe_load(open('./credentials.yml'))

account_list = []
for i in acct_credentials:
    account_list.append(i)
    
account_list=['GB']


for account in account_list:
    get_credentials = rc.sales_report_credentials(account)
    credentials = get_credentials[0]
    other_credentials = get_credentials[1]

    res_obj = rc.get_marketplace_res_obj(account, credentials)

    print('\n')
    print('Running report for ' + str(other_credentials['amz_account']))
    
    # ======== GET AUTH ========
    # Credentials for user created following the docs
    amw_client = boto3.client(
        'sts',
        aws_access_key_id=str(credentials['aws_access_key']),
        aws_secret_access_key=str(credentials['aws_secret_key']),
        region_name=str(other_credentials['region'])
    )
    # ROLE created following the docs
    # STS assume policy must be included in the role
    res = amw_client.assume_role(
        RoleArn=str(credentials['role_arn']),
        RoleSessionName='SellingPartnerAPI'
    )
    
    Credentials = res["Credentials"]
    AccessKeyId = Credentials["AccessKeyId"]
    SecretAccessKey = Credentials["SecretAccessKey"]
    SessionToken = Credentials["SessionToken"]
        
    aws_auth = AWSSigV4('execute-api',
                        aws_access_key_id=AccessKeyId,
                        aws_secret_access_key=SecretAccessKey,
                        aws_session_token=SessionToken,
                        region=str(other_credentials['region'])
                        )
    
    # ======== GET ACCESS TOKEN ======
    body = \
        {
            'grant_type': 'refresh_token',
            'client_id': str(credentials['lwa_app_id']),
            'refresh_token': str(credentials['refresh_token']),
            'client_secret': str(credentials['lwa_client_secret'])
        }
    
    h = {'Content-Type': 'application/json'}
    access_token_response = \
        requests.post('https://api.amazon.com/auth/o2/token', json=body, headers=h)
    access_token = access_token_response.json().get('access_token')
    
    
    # ======== CONSUME API ========
    request_url = str(other_credentials['endpoint']) + '/catalog/2020-12-01/items/B06XPMH32M?marketplaceIds=A1F83G8C2ARO7P'
    # request_url = str(other_credentials['endpoint']) + '/vendor/orders/v1/purchaseOrdersStatus?dataStartTime=2022-01-01T00:00:00.000000&dataEndTime=2022-01-31T00:00:00.000000'
    
    r = requests.get(
    request_url, auth=aws_auth, headers={'x-amz-access-token': access_token})
    
    # data = ast.literal_eval(str(resp))
    print('Response code: %d\n' % r.status_code)
    print(r.text)    
`


This is the error when requesting purchaseOrderStatus:

`Response code: 403

{
  "errors": [
    {
      "message": "The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.

The Canonical String for this request should have been
'GET
/vendor/orders/v1/purchaseOrdersStatus
dataEndTime=2022-01-31T00%3A00%3A00.000000&dataStartTime=2022-01-01T00%3A00%3A00.000000
host:sellingpartnerapi-eu.amazon.com
x-amz-access-token:xxx
x-amz-content-sha256:xxx
x-amz-date:20220307T161644Z
x-amz-security-token:xxx

host;x-amz-access-token;x-amz-content-sha256;x-amz-date;x-amz-security-token
xxx'

The String-to-Sign should have been
'AWS4-HMAC-SHA256
20220307T161644Z
20220307/eu-west-1/execute-api/aws4_request
xxx'
",
     "code": "InvalidSignature"
    }
  ]`

**Comment by timmyscj** [comment]
When I change the lwa_access_id, lwa_client_secret and refresh token to the app created for a US account, it works but pulls purchase orders for the US, not the specified country (GB). It doesn't make sense.
